src/Xamarin.Android.Build.Tasks/Utilities/ApplicationConfig.cs

```diff
@@ -55,6 +55,9 @@ sealed class ApplicationConfig
 		public uint   jni_remapping_replacement_type_count;
 		public uint   jni_remapping_replacement_method_index_entry_count;
 
+		// 3, for 4-byte alignment (4k memory pages); 15, for 16-byte alignment (16k memory pages)
```

Comment:
why are the mask 3 and 15 respecitvely?  noth 4 and 16? (silly question).

Reply:
With alignment to 4k, bits 0 and 1 of the address must be 0 and with 16k, bits 0-3 must be 0.  We could use the actual alignment here and then subtract `1` to obtain a mask from it, but it would require an additional instruction on the run time :) As it is now, we can **add** `1` to the mask to get the alignment value when reporting an error - an extra instruction but only in error situations, so performance isn't an issue.  Yes, I know, it's just one instruction :)

The relevant code is [here](https://github.com/dotnet/android/pull/9046/files#diff-bfc34e743751825ec3bd75052d5badba837e6f2746787f56bc3a092fd997aeacR63)